### PR TITLE
generated sequence

### PR DESCRIPTION
**Pull request by NikczemnyOkularnik**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: generate-sequence
🏅 Granted 5 XP for @NikczemnyOkularnik

**Comment by ziobron** [comment]
⏱ 3 XP extra for @NikczemnyOkularnik for on-time delivery

File: homework/generate-sequence/generateSequence.hpp [hunk 1]
```diff
@@ -1,7 +1,24 @@
 #pragma once
 #include <vector>
 
-std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
+std::vector<int> generateSequence(int count, int step)
+{
+    std::vector<int> afterGeneratedSequence;
```
Comment (nauka-programowania-MA) [inline homework/generate-sequence/generateSequence.hpp:6]: Reserve memory when you know how many elements will be inside vector.

File: homework/generate-sequence/generateSequence.hpp [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+
+    int temp = step;
+
+    for (int i = 1; i <= count; ++i)
+    {
+        afterGeneratedSequence.push_back(step);
+        step += temp;
+        //std::cout << countedElements.size() << " | ";
+        //std::cout << countedElements.capacity() << "\n";
```
Comment (nauka-programowania-MA) [inline homework/generate-sequence/generateSequence.hpp:15]: Remove

File: homework/generate-sequence/generateSequence.hpp [hunk 3]
```diff
@@ -1,7 +1,24 @@
...
+    }
+    afterGeneratedSequence.shrink_to_fit();
+
+    //std::cout << countedElements.size() << " | ";
+    //std::cout << countedElements.capacity() << "\n";
```
Comment (nauka-programowania-MA) [inline homework/generate-sequence/generateSequence.hpp:20]: Remove

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**